flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java

```diff
@@ -367,10 +367,24 @@ public static ResourceRequirements getResourceRequirements(
             Map<String, ExternalResource> externalResources,
             Map<String, String> externalResourceConfigKeys) {
         final Quantity cpuQuantity = new Quantity(String.valueOf(cpu));
-        final Quantity cpuLimitQuantity = new Quantity(String.valueOf(cpu * cpuLimitFactor));
+        final Quantity cpuLimitQuantity =
+                new Quantity(
+                        String.valueOf(
+                                getLimit(
+                                        cpu,
+                                        cpuLimitFactor,
+                                        resourceRequirements,
+                                        Constants.RESOURCE_NAME_CPU)));
         final Quantity memQuantity = new Quantity(mem + Constants.RESOURCE_UNIT_MB);
```

Comment:
It seems to be very inconsistent/wrong to take the value for limit from the podTemplate and not for the quantity.
I think we should make a decision and do it either for both or neither.